cluster create refactor

Orzelius opened the pull request:
Refactor the cluster creation code (once more) to assure easy maintainability and testability
* Create the following packages under `/create`:
	- `flags` containing the pflag.Value implementations
	- `clusterops` containing the options for cluster creation
		- `configmaker` containing the logic that aids in creation of talos and provision configuration for cluster creation
			- internal `makers` and `siderolinkbuilder` packages that hide the internal logic that configmaker uses
* Remove code duplication of default values. Now all default values come from the clusterops Get functions.
* Add unit tests for flag implementations.
* Add tests that compare machine configs generated for cluster create to default configs.
	These tests also functions as snapshot tests and will asure no undesired changes pass through in the future unnoticed.

Potential further improvements that could be made if judged beneficial enough:
* Split the dev and user-facing qemu creators. This would isolate our development related changes such that they don't cause issues with the user-facing dev command. If implemented, the split should be made such that the dev-qemu-maker extends the user-facing-qemu-maker, keeping us dogfooding the logic.
* Remove the purely CLI related options from clusterops (stuff like `StateDir`, `CniBinPath`, `TalosconfigDestination` etc)

part of [#11404](https://github.com/siderolabs/talos/issues/11404)

frezbo commented on cmd/talosctl/cmd/mgmt/cluster/create/clusterops/configmaker/internal/makers/common.go:
```diff
@@ -0,0 +1,600 @@
+
+const (
+	// gatewayOffset is the offset from the network address of the IP address of the network gateway.
+	gatewayOffset = 1
```
```suggestion
	gatewayOffset iota
```

Orzelius commented:
@frezbo Thank you for the feedback! I've implemented the changes in [a6bc0fb](https://github.com/siderolabs/talos/pull/11862/commits/a6bc0fba318f25834114981c6b90a07bd7a77634) so it's easier to review. I will squash it before merging. I also renamed the methods to make more sense and fixed a bug where the version contract was initialized too late causing latest configs to be generated even for older talos version.

Orzelius commented:
/m
